internal/rbd/rbd_util.go

```diff
@@ -1533,10 +1537,14 @@ func (rv *rbdVolume) setThickProvisioned() error {
 // the expansion can be allocated too.
 func (ri *rbdImage) isThickProvisioned() (bool, error) {
 	value, err := ri.GetMetadata(thickProvisionMetaKey)
-	if err != nil {
+	if err == librbd.ErrNotFound {
+		// check if the image is having deprecated metadata key.
+		value, err = ri.GetMetadata(deprecatedthickProvisionMetaKey)
```

Comment:
If we get here, it may be simpler to set the new key in any case. That way, only one check for the metadata is needed when the image is thin-provisioned.

It also helps with converting old images to the new metadata, and the deprecated key can be used only in this function (or ideally a new helper).

Reply:
@nixpanic if the error  is `err == librbd.ErrNotFound` and the metadata key `thickProvisionMetaKey` either it could be thin volume or it might be thick volume with old metadata set. For converting, yes if the value is "true" then we can set the metadata key again which will convert the old images to new metadata. Is that what you are looking for?

```suggestion
func (ri *rbdImage) isThickProvisioned() (bool, error) {
	value, err := ri.GetMetadata(thickProvisionMetaKey)
	if err == librbd.ErrNotFound {
		// check if the image is having deprecated metadata key.
		value, err = ri.GetMetadata(deprecatedthickProvisionMetaKey)
		if err == librbd.ErrNotFound {
			return false, nil
		}
		if value == "true" {
		      err=vi.setThickProvisioned(nil)
		      if err!=nil{
		            return false,err
		      }
		}...
	}
```

Reply:
When the image is thin-provisioned, setting `thickProvisionMetaKey` to `false` will prevent a 2nd lookup of the metadata on subsequent usage. That makes it also unneeded for `setThickProvisioned()` to check for a parent volume and makes the rest of this PR simpler.

Eventually we can set the metadata (when thin-provisioned) on image creation, if we decide that has a benefit.

Reply:
Yes, that makes sense but the only thing is we will be setting image metadata on all images irrespective it's thick or thin. Am fine either way let me know the suggestions. but still, we need to pass parent volume to `setThickProvisioned` to support backward compatibility?

Reply:
Setting the value would not hurt, so if it is not intentionally set (thin-provisioned), setting it to `false` is fine.

I do not think it will be needed to check if a parent volume has the metadata set when detecting the old key and setting the new one. I can not think of a reason why that is needed, as we already explicitly set the metadata when cloning or restoring from a snapshot.

Reply:
Still waiting on an update for this. There should not be need to check the parent  volume when getting/setting the thick-provisioned key.

Reply:
made changes as per suggestion.